### PR TITLE
#172 Add UI for launching m2sys easy scan exe and integrating

### PR DESCRIPTION
**Pull request by areklalo**

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/field/FingerprintM2sysFragmentController.java [hunk 1]
```diff
@@ -2,28 +2,94 @@
...
+import java.util.ArrayList;
 import java.util.List;
 
 /**
  * Created by user on 02.08.17.
  */
 public class FingerprintM2sysFragmentController {
 
-    public void controller() {
+    private static BiometricEngine biometricEngine;
```
Comment (pgesek) [inline omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/field/FingerprintM2sysFragmentController.java]: any reason for this being static?

File: omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/field/FingerprintM2sysFragmentController.java [hunk 2]
```diff
@@ -13,8 +17,13 @@ public void controller() {
...
 
-    public void enroll(@SpringBean("registrationCoreService") RegistrationCoreService service) {
+    public SimpleObject enroll(@SpringBean("registrationCoreService") RegistrationCoreService service) {
         BiometricEngine biometricEngine = service.getBiometricEngine();
-        biometricEngine.enroll(null);
+        BiometricSubject biometricSubject = new BiometricSubject();
+        BiometricSubject response = biometricEngine.enroll(biometricSubject);
+
+        SimpleObject result = new SimpleObject();
+        result.put("id",response.getSubjectId());
```
Comment (pgesek) [inline omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/field/FingerprintM2sysFragmentController.java]: space here

**Review by pgesek — CHANGES_REQUESTED** [review]

File: omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/field/FingerprintM2sysFragmentController.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package org.openmrs.module.registrationapp.fragment.controller.field;
+
+import org.openmrs.module.registrationcore.api.RegistrationCoreService;
+import org.openmrs.module.registrationcore.api.biometrics.BiometricEngine;
+import org.openmrs.ui.framework.annotation.SpringBean;
+
+/**
```
Comment (pgesek) [inline omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/field/FingerprintM2sysFragmentController.java]: Remove this

**Review by pgesek — CHANGES_REQUESTED** [review]

File: omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/field/FingerprintM2sysFragmentController.java (hunk 3 above)
Comment (areklalo) [reply omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/field/FingerprintM2sysFragmentController.java]: Corrected.

File: omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/field/FingerprintM2sysFragmentController.java (hunk 2 above)
Comment (areklalo) [reply omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/field/FingerprintM2sysFragmentController.java]: Corrected.

File: omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/field/FingerprintM2sysFragmentController.java (hunk 1 above)
Comment (areklalo) [reply omod/src/main/java/org/openmrs/module/registrationapp/fragment/controller/field/FingerprintM2sysFragmentController.java]: No I corrected this.

**Pull request merged**